objective.py

```diff
@@ -16,11 +18,42 @@ def set_data(self, X, y):
         self.X, self.y = X, y
 
     def compute(self, beta):
+        beta = np.array(beta).flatten()
```

Comment:
is it mandatory to cast to array and flatten ? We rather assume than beta is passed as as flat array already, and flatten in Solver.get_results()